File: examples/RubyExamples/testEndpoint.rb
# Copyright 2016 Ontotext AD
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# IMPORTANT!!! Please note the packages below, needed for this example IMPORTANT!!!
require "typhoeus"
require "json"


endpoint = "https://text.s4.ontotext.com/v1/"
api_key = "<your-credentials-here>"
key_secret = "<your-credentials-here>"

hydra = Typhoeus::Hydra.hydra
req = Typhoeus::Request.new(endpoint, userpwd: api_key + ":" + key_secret)
hydra.queue(req)
hydra.run
response = req.response

puts response.body, "\n"

# Response Code
print "Status Code: ", response.code, "\n\n"

# Response Headers
puts "Headers: "
response.headers.each do |type, header|
    print type, ": ", header, "\n"
end

File: examples/PythonExamples-requests/testEndpoint.py
# Copyright  2013, 2014, Ontotext AD
#
# This file is free software; you can redistribute it and/or modify it under
# the terms of the GNU Lesser General Public License as published by the Free
# Software Foundation; either version 2.1 of the License, or (at your option)
# any later version.
# This library is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
# details.
# You should have received a copy of the GNU Lesser General Public License
# along with this library; if not, write to the Free Software Foundation, Inc.,
# 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA

# IMPORTANT!!! Please install the packages below, needed for this example

import requests

endpoint = "https://text.s4.ontotext.com/v1"
api_key = "<your-credentials-here>"
key_secret = "<your-credentials-here>"

req = requests.get(endpoint, auth=(api_key, key_secret))

print (req.content.decode("utf-8"), "\n")

print ("Request Code: {}\n".format(req.status_code))

head = dict(req.headers)
print ("Headers: ")
for each in head:
    print (each.capitalize(), ": ", head[each])
